pkg/autoscaler/autoscaler/calculate/storage.go

```diff
@@ -0,0 +1,140 @@
+	}
+
+	// query total available storage size
+	resp := &Response{}
+	err := queryMetricsFromPrometheus(tac, client, availableSq, resp)
+	if err != nil {
+		return false, err
+	}
+	var availableSize uint64
+	for _, r := range resp.Data.Result {
+		if r.Metric.Cluster == tac.Spec.Cluster.Name {
+			availableSize, err = strconv.ParseUint(r.Value[1].(string), 10, 64)
+			if err != nil {
+				return false, err
+			}
+		}
+	}
+
+	// query total capacity storage size
+	resp = &Response{}
+	err = queryMetricsFromPrometheus(tac, client, capacitySq, resp)
+	if err != nil {
+		return false, err
+	}
+	var capacitySize uint64
+	for _, r := range resp.Data.Result {
+		if r.Metric.Cluster == tac.Spec.Cluster.Name {
+			capacitySize, err = strconv.ParseUint(r.Value[1].(string), 10, 64)
+			if err != nil {
+				return false, err
+			}
+		}
+	}
+	v := *metric.LeastRemainAvailableStoragePercent
+	baselineAvailableSize := (capacitySize / 100) * uint64(v)
+	storagePressure := false
+	if availableSize < baselineAvailableSize {
+		storagePressure = true
+	}
+
+	var newStatus, oldStatus *v1alpha1.MetricsStatus
+	for _, m := range tac.Status.TiKV.MetricsStatusList {
+		if m.Name == string(corev1.ResourceStorage) {
+			oldStatus = &m
+			break
+		}
+	}
+	storageMetrics := v1alpha1.StorageMetricsStatus{
+		StoragePressure:          pointer.BoolPtr(storagePressure),
+		AvailableStorage:         pointer.StringPtr(byteCountDecimal(availableSize)),
+		CapacityStorage:          pointer.StringPtr(byteCountDecimal(capacitySize)),
+		BaselineAvailableStorage: pointer.StringPtr(byteCountDecimal(baselineAvailableSize)),
+	}
+	if oldStatus != nil {
+		oldStatus.StoragePressure = storageMetrics.StoragePressure
+		oldStatus.AvailableStorage = storageMetrics.AvailableStorage
+		oldStatus.CapacityStorage = storageMetrics.CapacityStorage
+		oldStatus.BaselineAvailableStorage = storageMetrics.BaselineAvailableStorage
+		newStatus = oldStatus
+	} else {
+		newStatus = &v1alpha1.MetricsStatus{
+			Name:                 string(corev1.ResourceStorage),
+			StorageMetricsStatus: storageMetrics,
+		}
+	}
+
+	if storagePressure {
+		if !isStoragePressureStartTimeRecordAlready(tac.Status) {
+			newStatus.StorageMetricsStatus.StoragePressureStartTime = &metav1.Time{Time: time.Now()}
+		}
+	} else {
+		newStatus.StoragePressureStartTime = nil
+	}
+	addMetricsStatusIntoMetricsStatusList(*newStatus, &tac.Status.TiKV.BasicAutoScalerStatus)
+	return storagePressure, nil
+}
+
+// TODO: add unit test
+func isStoragePressureStartTimeRecordAlready(tacStatus v1alpha1.TidbClusterAutoSclaerStatus) bool {
+	if tacStatus.TiKV == nil {
+		return false
+	}
+	if len(tacStatus.TiKV.MetricsStatusList) < 1 {
+		return false
+	}
+	for _, metricsStatus := range tacStatus.TiKV.MetricsStatusList {
+		if metricsStatus.Name == "storage" {
+			if metricsStatus.StoragePressureStartTime != nil {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func byteCountDecimal(b uint64) string {
```

Comment:
is it possible to use [humanize.Bytes](https://pkg.go.dev/github.com/dustin/go-humanize?tab=doc#Bytes)? we already used this in other places.

Reply:
sure, updated.